sound/soc/sof/sof-audio.c

```diff
@@ -92,11 +129,20 @@ static int sof_widget_free(struct snd_sof_dev *sdev, struct snd_sof_widget *swid
 		.id = swidget->comp_id,
 	};
 	struct sof_ipc_reply reply;
-	int ret;
+	int ret = 0;
+
+	/* decrement usecount and update config for static DAI widgets */
+	if (swidget->use_count > 1) {
+		if (!swidget->dynamic_pipeline_widget)
+			ret = sof_static_widget_config(sdev, swidget);
```

Comment:
maybe this isnt easy to see but before this patch I had a function to just send the DAI_CONFIG in sof_widget_list_free(). So for statc pipelines, all that  you'd do during hw_free, is send the DAI_CONFIG IPC for all DAI widgets. But with this patch, you'd still do the same except, that you'd do that through call sof_widget_free() which for static pipelines will always end up here because a widget's use_count for static pipelines would never go dowm below 1.make sense?

Reply:
Right, my question is: is the `DAI_CONFIG` IPC also sent now before your PR from `hw_free`? It looks like now we're only sending `SOF_IPC_STREAM_PCM_FREE`?

Reply:
> `DAI_CONFIG`? After your PR static pipelines will also send `DAI_CONFIG` from `.hw_free()`? Is this done without your PR now too?

Yes, it is done for ALH and HDA DAIs during BE hw_free.

Reply:
> As for my original question, I think the answer would be - yes, this is the freeing path, so it seems correct to send IPCs like `SOF_IPC_TPLG_COMP_FREE`, but I'm still not quite sure I understand whether they would be sent multiple times for widgets with `.use_count > 1`?

Ahh yes, you are right. For static widgets, it should be sent when use_count is exactly 1. Let me fix this